pennylane/optimize/spsa.py

```diff
@@ -270,7 +271,8 @@ def compute_grad(self, objective_fn, args, kwargs):
                 shots = Shots(objective_fn.device._raw_shot_sequence)  # pragma: no cover
             else:
                 shots = Shots(None)
-            if np.prod(objective_fn.func(*args).shape(objective_fn.device, shots)) > 1:
+
+            if np.prod(objective_fn.func(*args, **kwargs).shape(objective_fn.device, shots)) > 1:
```

Comment:
I'm wondering: why exactly do we need another evaluation here? Couldn't we infer this bit of information from the computed values `yplus` or `yminus`? :thinking:

Reply:
Hmmm... I'm not sure! My intuition says no based on the fact that `yminus` and `yplus` are parameter-shifts of `objective_fn`, and `objective_fn` isn't linear in general.

Reply:
But we're only extracting the shape, which should not depend on the parameter position, right? :sweat_smile:

Reply:
Right! I missed that 😅. I think for now let's just leave it since I'm slightly out of my league here 😁